src/transformers/models/bart/modeling_tf_bart.py

```diff
@@ -963,7 +963,8 @@ def call(
             else self.config.output_hidden_states
         )
         return_dict = inputs["return_dict"] if inputs["return_dict"] is not None else self.config.return_dict
-        if not use_cache:
+
+        if not use_cache or inputs["decoder_input_ids"].shape[-1] > 1:
```

Comment:
This was actually not correct neither in PT nor in TF. If `use_cache=True` => this does not mean that `decoder_input_ids.shape[-1] == 1`. It can very well be that `decoder_input_ids` have more than one token in which case we do need to generate a causal mask.